arduino-ide-extension/src/browser/boards/boards-config-component.tsx

```diff
@@ -0,0 +1,328 @@
+        {!selected ? (
+          ''
+        ) : (
+          <div className="selected-icon">
+            <i className="fa fa-check" />
+          </div>
+        )}
+      </div>
+    );
+  }
+
+  private readonly onClick = () => {
+    this.props.onClick(this.props.item);
+  };
+}
+
+export class BoardsConfigComponent extends React.Component<
+  BoardsConfigComponent.Props,
+  BoardsConfigComponent.State
+> {
+  private readonly toDispose: DisposableCollection;
+
+  constructor(props: BoardsConfigComponent.Props) {
+    super(props);
+    this.state = {
+      searchResults: [],
+      showAllPorts: false,
+      query: '',
+    };
+    this.toDispose = new DisposableCollection();
+  }
+
+  override componentDidMount(): void {
+    this.toDispose.pushAll([
+      this.props.onAppStateDidChange(async (state) => {
+        if (state === 'ready') {
+          const searchResults = await this.queryBoards({});
+          this.setState({ searchResults });
+        }
+      }),
+      this.props.notificationCenter.onPlatformDidInstall(() =>
+        this.updateBoards(this.state.query)
+      ),
+      this.props.notificationCenter.onPlatformDidUninstall(() =>
+        this.updateBoards(this.state.query)
+      ),
+      this.props.notificationCenter.onIndexUpdateDidComplete(() =>
+        this.updateBoards(this.state.query)
+      ),
+      this.props.notificationCenter.onDaemonDidStart(() =>
+        this.updateBoards(this.state.query)
+      ),
+      this.props.notificationCenter.onDaemonDidStop(() =>
+        this.setState({ searchResults: [] })
+      ),
+      this.props.onFilteredTextDidChangeEvent((query) => {
+        if (typeof query === 'string') {
+          this.setState({ query }, () => this.updateBoards(this.state.query));
+        } else {
+          const action = query?.action;
+          if (action) {
+            const currentQuery = this.state.query;
+            if (!currentQuery && action === 'clear-if-not-empty') {
+              return;
+            }
+            this.setState({ query: '' }, () =>
+              this.updateBoards(this.state.query)
+            );
+          }
+        }
+      }),
+    ]);
+  }
+
+  override componentWillUnmount(): void {
+    this.toDispose.dispose();
+  }
+
+  private readonly updateBoards = (
+    eventOrQuery: React.ChangeEvent<HTMLInputElement> | string = ''
+  ) => {
+    const query =
+      typeof eventOrQuery === 'string'
+        ? eventOrQuery
+        : eventOrQuery.target.value.toLowerCase();
+    this.setState({ query });
+    this.queryBoards({ query }).then((searchResults) =>
+      this.setState({ searchResults })
+    );
+  };
+
+  private readonly queryBoards = (
+    options: { query?: string } = {}
+  ): Promise<Array<BoardWithPackage>> => {
+    return this.props.searchBoards(options);
+  };
+
+  private readonly toggleFilterPorts = () => {
+    this.setState({ showAllPorts: !this.state.showAllPorts });
```

Comment:
here given the new state prop `showAllPorts` is dependent on the previous I think an updater function may be more approriate "if you need to compute values based on the current state": [reference](https://legacy.reactjs.org/docs/faq-state.html#how-do-i-update-state-with-values-that-depend-on-the-current-state)

```
this.setState((state) => ({ showAllPorts: !state.showAllPorts }));
```

Reply:
Thanks for the hint. As I mentioned in another comment (https://github.com/arduino/arduino-ide/pull/2165#discussion_r1287029159), the purpose of this PR is unrelated to tweaking the React code. This is an existing code:

https://github.com/arduino/arduino-ide/blob/144df893d0dafec64a26565cf912a98f32572da9/arduino-ide-extension/src/browser/boards/boards-config.tsx#L244

I do not have the capacity to fix existing UI flaws unless they're breaking.